### ISSUE TITLE
Can nextMonotonic work under high load?

### ISSUE THREAD

**Issue by martinscheffler** [paradoxical-io/cassieq#126]
nextMonotonic() fetches the current monotonic from cassandra, then tries to write the next higher monotonic back to cassandra. If in the meantime the value was changed by a concurrently running client then it retries this until success.
So when we have a large number of concurrent writers all trying to obtain a monoton then all except one will fail to get the valid next one. Isn't this a fundamental performance problem? Even writing to the same queue from multiple threads will become very problematic when latency to the cassandra db is high

**Comment by devshorts** [comment]
Yes, it's probably going to be an issue. We had discussed a few other possibilities of dealing with uniqueness, whether plugging in a supporting system to generate the monotons (like redis) or moving to time based uuids for message identifiers. 

Doing the former is the easiest win even though it violates one of our original design philosophies of no external infra but Cassandra. 

Doing the latter requires rework of how we deal with bucket completion.

However, we did have an idea of how to provide better invisibility timeout support (without blocking later messages) using a delayed time based bucket. if we did this it may solve all these issues
